comp/netflow/goflowlib/convert.go

```diff
@@ -60,3 +67,79 @@ func convertFlowType(flowType flowpb.FlowMessage_FlowType) common.FlowType {
 	}
 	return flowTypeStr
 }
+
+func applyAdditionalFields(flow *common.Flow, additionalFields common.AdditionalFields) {
+	for destination, fieldValue := range additionalFields {
+		applied := applyAdditionalField(flow, destination, fieldValue)
+		if applied {
+			// We replaced a field of common.Flow with an additional field, no need to keep it in the map
+			delete(additionalFields, destination)
+		}
+	}
+	flow.AdditionalFields = additionalFields
+}
+
+func applyAdditionalField(flow *common.Flow, destination string, fieldValue any) bool {
+	// Make sure FlowFieldsTypes includes the type of the following fields
+	switch destination {
+	case "direction":
+		setValue(&flow.Direction, fieldValue)
+	case "start":
+		setValue(&flow.StartTimestamp, fieldValue)
+	case "end":
+		setValue(&flow.EndTimestamp, fieldValue)
+	case "bytes":
+		setValue(&flow.Bytes, fieldValue)
+	case "packets":
+		setValue(&flow.Packets, fieldValue)
+	case "ether_type":
+		setValue(&flow.EtherType, fieldValue)
+	case "ip_protocol":
+		setValue(&flow.IPProtocol, fieldValue)
+	case "exporter.ip":
+		setValue(&flow.ExporterAddr, fieldValue)
+	case "source.ip":
+		setValue(&flow.SrcAddr, fieldValue)
+	case "source.port":
+		var port uint64
+		setValue(&port, fieldValue)
+		flow.SrcPort = int32(port)
+	case "source.mac":
+		setValue(&flow.SrcMac, fieldValue)
+	case "source.mask":
+		setValue(&flow.SrcMask, fieldValue)
+	case "destination.ip":
+		setValue(&flow.DstAddr, fieldValue)
+	case "destination.port":
+		var port uint64
+		setValue(&port, fieldValue)
+		flow.DstPort = int32(port)
+	case "destination.mac":
+		setValue(&flow.DstMac, fieldValue)
+	case "destination.mask":
+		setValue(&flow.DstMask, fieldValue)
+	case "ingress.interface":
+		setValue(&flow.InputInterface, fieldValue)
+	case "egress.interface":
+		setValue(&flow.OutputInterface, fieldValue)
+	case "tcp_flags":
+		setValue(&flow.TCPFlags, fieldValue)
+	case "next_hop.ip":
+		setValue(&flow.NextHop, fieldValue)
+	case "tos":
+		setValue(&flow.Tos, fieldValue)
+	default:
+		return false
+	}
+	return true
+}
+
+type flowFieldsTypes interface {
+	uint64 | uint32 | []byte
+}
+
+func setValue[T flowFieldsTypes](field *T, value any) {
+	if v, ok := value.(T); ok {
+		*field = v
+	}
```

Comment:
Yeah that's a good point, with this if the type is incorrect in the configuration, e.g
```
- field: 8
  destination: "destination.port"
  type: "string" # incorect type for this field (port)
```
we will indeed ignore the error
I'm looking at how we could log this, but I don't know if it will be feasible without drilling the logger through all functions

Reply:
Addressed in 6f0f0d0
Changed so that when additional fields destination is set to one of our default fields (e.g. `destination.port`, `packets`, ...) we don't rely on the `type` configuration (and it is not needed). setValue should therefore not fail

It also simplifies configuration from the user side
```
- field: 8
  destination: "destination.port"
```

Reply:
If the user still attempts to configure it with the wrong type though, doesn't that mean we'll just ignore it? I feel like we should have something to indicate to the user that their config is bad and will be ignored

Reply:
Indeed, do you think failing on startup is a better option? This is what we do on config error for netflow right now ([here](https://github.com/DataDog/datadog-agent/blob/main/comp/netflow/config/config.go#L62))

Reply:
Added a warn log on startup when type is invalid in d7815ea